does not work in collaboration with `@pytest.mark.flaky`

hello there, I think there is something about `pytest-asyncio-cooperative` that skips tests from rerunning, I have tried with both [pytest-rerunfailures](https://github.com/pytest-dev/pytest-rerunfailures), [flaky](https://github.com/box/flaky) and [pytest-retry](https://github.com/str0zzapreti/pytest-retry), but none of the three libraries work in conjunction with `pytest-asyncio-cooperative`, on the output they say there was a rerun, but actually no rerun happens, maybe `pytest-asyncio-cooperative` caches the result or something and just repeats them

For example:

```
@pytest.mark.asyncio_cooperative
@pytest.mark.flaky()
async def test_example():
    print("Running test_example")
    assert False
```

Prints `Running test_example` only once, while without `asyncio_cooperative` it prints twice, even though the output says:

```
=============================================================================== short test summary info ===============================================================================
FAILED tests/test_simple_assertion.py::test_example - assert False
======================================================================= 1 failed, 34 warnings, 1 rerun in 1.15s =======================================================================
```

the retry libraries seem to be working correctly, but the retries run "instantly"

--- a/pytest_asyncio_cooperative/plugin.py
+++ b/pytest_asyncio_cooperative/plugin.py
@@ -304,10 +304,17 @@ def pytest_runtestloop(session):
                         result = None
                         def run_in_thread():
                             nonlocal result
-                            result = asyncio.run(new_task)
+                            try:
+                                result = asyncio.run(new_task)
+                            except Exception as e:
+                                result = e
                         thread = threading.Thread(target=run_in_thread)
                         thread.start()
                         thread.join()
+
+                        if isinstance(result, Exception):
+                            raise result # type: ignore
+
                         return result
 
                     item.runtest = sync_wrapper

--- a/pytest_asyncio_cooperative/plugin.py
+++ b/pytest_asyncio_cooperative/plugin.py
@@ -1,3 +1,4 @@
+import threading
 import asyncio
 import collections.abc
 import functools
@@ -290,7 +291,30 @@ def pytest_runtestloop(session):
                         item_by_coro[new_task] = item
                         continue
 
-                item.runtest = lambda: result.result()
+                # We need to change .runtest to a synchronous function for pytest
+                # however, if it is called again by retry libraries we need to rerun
+                # the test instead of retuning the previous result
+                def wrap_in_sync():
+                    def sync_wrapper():
+                        new_task = item_to_task(item)
+                        
+                        # We use a new thread because we can't block for an async function
+                        # in the same thread as the current running event loop, nor
+                        # we can nest event loops
+                        result = None
+                        def run_in_thread():
+                            nonlocal result
+                            result = asyncio.run(new_task)
+                        thread = threading.Thread(target=run_in_thread)
+                        thread.start()
+                        thread.join()
+                        return result
+
+                    item.runtest = sync_wrapper
+
+                    return result.result()
+
+                item.runtest = wrap_in_sync
 
                 item.ihook.pytest_runtest_protocol(item=item, nextitem=None)
 

